RELEASING.md

```diff
@@ -0,0 +1,84 @@
+Releasing a new version
+=======================
+
+We will use the `v0.5.3` release (#738) as an example of how to release a new
+FDO version:
+
+* Fork the repo and create a new branch for the new release:
+
+    ```bash
+    gh repo fork fdo-rs/fido-device-onboard-rs --clone --remote
+    git pull upstream main
+    git checkout -b prepare-v0.5.3
+    ```
+
+* Update the `fido-device-onboard.spec` file and set the new version: `Version: 0.5.3`
+* Update the `version` within the `[[package]]` section in all the `Cargo.toml` files.
+* Update the `[[dependencies]]` in all the `Cargo.toml` files to use the latest
+FDO versions.
+* Update the `libfdo-data/fdo_data.h` file to reflect the correct version:
+
+    ```c
+    #define FDO_DATA_MAJOR 0
+    #define FDO_DATA_MINOR 5
+    #define FDO_DATA_PATCH 3
+    ```
+
+* Update `Cargo.lock` file:
+
+    ```bash
+    cargo update --offline --workspace
+    ```
+
+* Commit all the changes and create a PR (see #738 with all the changes described
+above):
+
+    ```bash
+    git add *
+    git commit -s -m "chore: bump for 0.5.3 release" -m "Prepare for the 0.5.3 release."
+    gh pr create
+    ```
+
+* Once all the tests passed, tag and sign the release:
+
+    ```bash
+    git tag -a -s v0.5.3
+    ```
+
+* Create the release in the github project by selecting the just created
+`v0.5.3` tag.
+
+Downstream releases
+-------------------
+
+* Fedora releases are handled by packit: A new bodhi update will be generated
+each time a release is created in github.
+
+* CentOS releases must be triggered manually:
+
+  * Create personal access tokens in github and gitlab and use
+    the `~/.config/packit.yaml` file to configure the packit CLI:
+
+      ```yaml
+      ---
+      authentication:
+        github.com:
+          token: 
+        gitlab.com:
+          token: 
+      ```
+
+  * Create the required Jira Bug that will be attached to the MR
+  * Propose the changes downstream (only `c10s` and `c9s` dist-git branches are supported):
+
+      ```bash
+      packit propose-downstream --pr \
+                                --dist-git-branch c10s \
+                                --resolve-bug RHEL-00000 \
+                                --package fido-device-onboard-centos \
+                                v0.5.3
+      ```
+
+  * Check all the tests passed in the corresponding [MR](https://gitlab.com/redhat/centos-stream/rpms/fido-device-onboard/-/merge_requests)
+  * Generate a build with `centpkg`
+  * Attach the Errata
```

Comment:
I think all the centos stream bits can only be done by RHers. I think that should be noted.